thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -89,6 +90,7 @@ private void setupTask(TaskInfo taskInfo, TaskContext taskContext) throws Except
     jobExecutionId = detectionTaskInfo.getJobExecutionId();
     anomalyFunctionFactory = taskContext.getAnomalyFunctionFactory();
     anomalyFunction = anomalyFunctionFactory.fromSpec(anomalyFunctionSpec);
+    ruleBasedAnomalyFunction = anomalyFunctionFactory.getRuleBasedAnomalyFunction(anomalyFunctionSpec);
```

Comment:
instead of this, we can have anomalyFunctionFactory return a list of anomalyFunctions and we call the analyze method for each anomalyFunction and the only code change required would be in anomalyFunctionFactory

Reply:
Sure. Will change to a list of anomaly functions.